pkg/apb/bind.go

```diff
@@ -86,8 +83,24 @@ func getPodName(output []byte, log *logging.Logger) (string, error) {
 	return podname[len(podname)-1], nil
 }
 
-func monitorOutput(podname string) ([]byte, error) {
-	return runCommand("oc", "logs", "-f", podname)
+func monitorOutput(podname string, mon chan []byte, log *logging.Logger) {
+	// TODO: Error handling here
+	// It would also be nice to gather the script output that exec runs
+	// instead of only getting the credentials
+	retries := 5
+	for r := 1; r <= retries; r++ {
+		output, _ := runCommand("oc", "exec", podname, "broker-bind-creds")
+
+		if strings.Contains(string(output), "BIND_CREDENTIALS") {
+			mon <- output
+			close(mon)
+			return
+		}
+
+		time.Sleep(time.Duration(r) * time.Second)
+		log.Warning("Retry attempt %d: exec into %s failed", r, podname)
+	}
+	return
```

Comment:
Need to send a timeout error message to the channel and ```close(mon)```. Return here isn't enough.